master/internal/db/postgres_tasks.go

```diff
@@ -98,14 +98,19 @@ WHERE task_id = $1
 func (db *PgDB) AddAllocation(a *model.Allocation) error {
 	return db.namedExecOne(`
 INSERT INTO allocations
-	(task_id, allocation_id, slots, resource_pool, start_time, state)
+	(task_id, allocation_id, slots, resource_pool, start_time, state, dtrain_port, 
+	inter_train_process_comm_port1, inter_train_process_comm_port2,c10d_port)
```

Comment:
i think we should have a `map[string]int` (or a list of strings) of ports instead of storing each port in its own column, individually.